ls1intum/Artemis
# `Development`: Refactor resource loader service

### Checklist
#### General


#### Server
#### Changes affecting Programming Exercises

### Motivation and Context
While implementing #6273 I noticed that the resource loader service uses `String`s to handle paths.

### Description
This PR refactors this service to use proper `Path` objects to handle paths.

It also fixes an issue introduced in #6273 where invalid paths are created on Windows systems (`*`s are not allowed): The file patterns have now been moved to a separate parameter for the `getResources` method.

By extension the `FileService::copyResources` had to be refactored, too, as it one could not handle the paths converted to strings on Windows systems.

### Steps for Testing

Prerequisites:
- 1 Instructor

1. Create a new programming exercise of any type.
2. The template/solution/test repositories should be filled with the expected files.

### Review Progress






#### Code Review
#### Manual Tests

### Test Coverage



| Class/File | Branch | Line |
|------------|-------:|-----:|
| ResourceLoaderService.java | 92% | 94% |
